### ISSUE TITLE
Small integer / large integer division should emit a warning

### ISSUE THREAD

**Issue by sharazam** [rust-lang-nursery/rust-clippy#1949]
Small integer / larger integer always gives you 0 with no warning at all. Happened to me by accident. For integer types, this should be a lint, because it is always 0. Take this code: 

```rust
fn main() {
    let color_red: u8 = 72;
    let color_max: u8 = 255;
    let color_percentage = color_red / color_max;
    println!("{:?}", color_percentage); // 0
}
```

Yes I know, stupid me, but I simply wondered why my texture was not visible. It could be the reason of many things, ended up being an integer division thing.

**Comment by sharazam** [comment]
After discussion, this is likely not feasible because runtime variables cannot be checked, only constants.
